slides/bitcoin_network.tex

```diff
@@ -0,0 +1,171 @@
+% Choose one to switch between slides and handout
+%\documentclass[]{beamer}
+\documentclass[handout]{beamer}
+
+% Video Meta Data
+\title{Bitcoin, Blockchain and Cryptoassets}
+\subtitle{The Bitcoin Network}
+\author{Prof. Dr. Fabian Schär}
+\institute{University of Basel}
+
+% Config File
+\input{../config/config.tex}
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+\begin{document}
+
+\thispagestyle{empty}
+\begin{frame}[noframenumbering]
+	\titlepage
+\end{frame}
+
+%%%
+\begin{frame}{Bitcoin is a Peer-to-Peer Network}
+	\centering
+	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
+		\input{../assets/figures/p2p}
+		\color{focus}
+		\node (text) at (-0.5,3.5) { Nodes };
+		\draw[->, thick]	(text.south) -- (agenta.west);
+		\draw[->, thick]	(text.south) -- (agentb.north west);
+	\end{tikzpicture}
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{Joining the Network}
+	\begin{center}
+		\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
+			\input{../assets/figures/join_network}
+		\end{tikzpicture}
+	\end{center}
+
+	\vspace{1cm}
+	
+	After the initial handshake, we request a list of addresses and connect to more peers.
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{Functionality of a Full Node}
+	\centering
+	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
+		\input{../assets/figures/node_functions}
+	\end{tikzpicture}
+\end{frame}
+%%%
+
+
+%%%
+\begin{frame}{Verify New Blocks}
+	
+	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
+		\input{../assets/figures/full_node_functions}
+	\end{tikzpicture}
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{Distribute Verified Blocks}
+	
+	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
+		\input{../assets/figures/full_node_functions2}
+	\end{tikzpicture}
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{Wallet Functionality}
+	\centering
+	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
+		\input{../assets/figures/wallet_functions}
+	\end{tikzpicture}
+\end{frame}
+%%%
+
+\begin{frame}{Mining Functionality}
+	\centering
+	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
+		\input{../assets/figures/mining_functions}
```

Comment:
Create block -> Assemble Blocks
Also, please change order, i.e. assemble blocks in the left and compute on the right.